pkg/operator/ceph/cluster/osd/topology.go

```diff
@@ -63,31 +63,17 @@ func extractTopologyFromLabels(labels map[string]string) (map[string]string, str
 	// not including the host name
 	var topologyAffinity string
 
-	// check for the region k8s topology label that was deprecated in 1.17
+	// check for the region k8s topology label
 	const regionLabel = "region"
-	region, ok := labels[corev1.LabelZoneRegion]
-	if ok {
-		topology[regionLabel] = region
-		topologyAffinity = formatTopologyAffinity(corev1.LabelZoneRegion, region)
-	}
-
-	// check for the region k8s topology label that is GA in 1.17.
-	region, ok = labels[corev1.LabelZoneRegionStable]
+	region, ok := labels[corev1.LabelZoneRegionStable]
 	if ok {
 		topology[regionLabel] = region
 		topologyAffinity = formatTopologyAffinity(corev1.LabelZoneRegionStable, region)
 	}
 
-	// check for the zone k8s topology label that was deprecated in 1.17
+	// check for the zone k8s topology label
 	const zoneLabel = "zone"
-	zone, ok := labels[corev1.LabelZoneFailureDomain]
-	if ok {
-		topology[zoneLabel] = zone
-		topologyAffinity = formatTopologyAffinity(corev1.LabelZoneFailureDomain, zone)
-	}
-
-	// check for the zone k8s topology label that is GA in 1.17.
-	zone, ok = labels[corev1.LabelZoneFailureDomainStable]
+	zone, ok := labels[corev1.LabelZoneFailureDomainStable]
```

Comment:
Ditto. Even though the label is deprecated, support is not fully removed. Kubernetes docs still include them in their well-known labels list: https://kubernetes.io/docs/reference/labels-annotations-taints/